File: SampleApp/Objective-C/SampleApp/Native/NativeTableViewController.m
//
//  NativeTableViewController.m
//  SampleApp
//
//  Copyright © 2017年 fluct, Inc. All rights reserved.
//

#import "NativeTableViewController.h"
@import FluctSDK;

@implementation NativeTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.tableView.estimatedRowHeight = 50;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 50;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    if (indexPath.row % 20 == 0) {
        FSSNativeTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"native" forIndexPath:indexPath];
        [cell loadAdWithGroupID:@"1000076934" unitID:@"1000115021"];
        return cell;

    } else {
        UITableViewCell *otherCell = [tableView dequeueReusableCellWithIdentifier:@"other"];
        return otherCell;
    }
}

@end

File: SampleApp/Swift/SDK/SampleApp-Swift/Native/NativeTableViewController.swift
//
//  NativeTableViewController.swift
//  SampleApp-Swift
//
//  Copyright © 2017年 fluct, Inc. All rights reserved.
//

import UIKit
import FluctSDK

class NativeTableViewController: UITableViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.tableView.estimatedRowHeight = 50
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 50
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        if indexPath.row % 20 == 0 {
            let cell: FSSNativeTableViewCell = tableView.dequeueReusableCell(withIdentifier: "native", for: indexPath) as! FSSNativeTableViewCell
            cell.loadAd(withGroupID: "1000076934", unitID: "1000115021")
            return cell
        } else {
            let otherCell: UITableViewCell = tableView.dequeueReusableCell(withIdentifier: "other", for: indexPath)
            return otherCell
        }
    }
}
